testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,263 @@
+            if (skipIteration) {
+                continue;
+            }
+
+            final Object[] parameterValues = getParameterValues(parameterTypes);
+
+            final boolean isCorrect = invokeAndCheck(method, parameterValues, parameterTypes);
+            if (!isCorrect) {
+                return false;
+            }
+
+        }
+        return true;
+    }
+
+    private Method[] getAccessibleMethods() {
+        final Method[] declaredMethods = utilClass.getDeclaredMethods();
+        final List<Method> methodList = newLinkedList();
+
+        for (Method method : declaredMethods) {
+            final boolean isPrivate = Modifier.isPrivate(method.getModifiers());
+            if (!isPrivate) {
+                methodList.add(method);
+            }
+        }
+
+        final Method[] result = methodList.toArray(new Method[methodList.size()]);
+        return result;
+    }
+
+    private Object[] getParameterValues(Class[] parameterTypes) {
+        final Object[] parameterValues = new Object[parameterTypes.length];
+        for (int i = 0; i < parameterTypes.length; i++) {
+            final Class type = parameterTypes[i];
+            parameterValues[i] = defaultValuesMap.get(type);
+        }
+        return parameterValues;
+    }
+
+    private boolean invokeAndCheck(Method method, Object[] parameterValues, Class[] parameterTypes) {
+        for (int i = 0; i < parameterValues.length; i++) {
+            Object[] copiedParametersArray = Arrays.copyOf(parameterValues, parameterValues.length);
+            final boolean isPrimitive = primitiveValues.contains(parameterTypes[i]);
+            if (!isPrimitive) {
+                copiedParametersArray[i] = null;
+            }
+
+            final boolean isCorrect = invokeAndCheck(method, copiedParametersArray);
+
+            if (!isCorrect) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean invokeAndCheck(Method method, Object[] params) {
+        try {
+            method.invoke(null, params);
+        } catch (InvocationTargetException ex) {
+            boolean isValid = validateException(method.getName(), ex);
+            return isValid;
+        } catch (IllegalAccessException e) {
+            throw Exceptions.wrappedCause(e);
+        }
+        return false;
+    }
+
+    private boolean validateException(String methodName, InvocationTargetException ex) {
+        final Throwable cause = ex.getCause();
+        checkException(cause);
+        final boolean result = isCorrectStackTraceElements(methodName, cause);
+        return result;
+    }
+
+    private void checkException(Throwable cause) {
+        final boolean isCorrectException = cause instanceof NullPointerException;
+        if (!isCorrectException) {
+            throw Exceptions.wrappedCause(cause);
+        }
+    }
+
+    private boolean isCorrectStackTraceElements(String methodName, Throwable cause) {
```

Comment:
Please be more specific in the method names like this one (i.e. `correct` seems too broad).  
Also, please have a JavaDoc for these methods.